Mathlib/FieldTheory/IntermediateField/Adjoin/Algebra.lean

```diff
@@ -24,6 +24,32 @@ variable (F : Type*) [Field F] {E : Type*} [Field E] [Algebra F E] (S : Set E)
 theorem algebra_adjoin_le_adjoin : Algebra.adjoin F S ≤ (adjoin F S).toSubalgebra :=
   Algebra.adjoin_le (subset_adjoin _ _)
 
+namespace algebraAdjoinAdjoin
+
+/-- `IntermediateField.adjoin` as an algebra over `Algebra.adjoin`. -/
+scoped instance : Algebra (Algebra.adjoin F S) (adjoin F S) :=
+  (Subalgebra.inclusion <| algebra_adjoin_le_adjoin F S).toAlgebra
+
+scoped instance (X) [SMul X F] [SMul X E] [IsScalarTower X F E] :
+    IsScalarTower X (Algebra.adjoin F S) (adjoin F S) :=
+  Subalgebra.inclusion.instIsScalarTowerSubtypeMem (algebra_adjoin_le_adjoin F S) _
+
+scoped instance (X) [MulAction E X] : IsScalarTower (Algebra.adjoin F S) (adjoin F S) X :=
+  Subalgebra.inclusion.instIsScalarTowerSubtypeMem_1 (algebra_adjoin_le_adjoin F S) _
```

Comment:
If `Subalgebra.inclusion.instIsScalarTowerSubtypeMem_1` is intended to be referenced elsewhere. I think we should give it a proper name.

Reply:
I now use `Bot` and `Top` to distinguish the two instances, see [e22c63c](https://github.com/leanprover-community/mathlib4/pull/24004/commits/e22c63c3b108034a1652d7cafc7cf8fee6ba14a8).